### PR TITLE
Fix #7089: SelectOne Refactor findSelectItem

### LINKED ISSUE #7089: SelectOneMenu/CascadeSelect: Refactor findSelectItem
**Description**
`SelectOneMenu.findEditableValue` and `CascadeSelect.findItemByValue` are the same basic method.  This should be refactored in 11.0 up to the parent `SelectRenderer`.

### PR DESCRIPTION
**Pull request by melloware**
We already have Integration Tests checking both scenarios so we can verify this refactor

### CODE REVIEW

File: src/main/java/org/primefaces/renderkit/SelectOneRenderer.java [hunk 1]
```diff
@@ -71,5 +77,48 @@ protected Object getSubmittedValues(UIComponent component) {
...
+                String valueOrLabel, boolean byValue) {
+        SelectItem foundValue = null;
+        for (int i = 0; i < selectItems.size(); i++) {
+            SelectItem item = selectItems.get(i);
+            if (item instanceof SelectItemGroup) {
+                SelectItemGroup selectItemGroup = (SelectItemGroup) item;
+                if (selectItemGroup.getSelectItems() == null) {
+                    continue;
+                }
+                foundValue = findSelectItem(fc, component,  converter, Arrays.asList(selectItemGroup.getSelectItems()), valueOrLabel, byValue);
```
Comment (tandraschko) [inline src/main/java/org/primefaces/renderkit/SelectOneRenderer.java:128]: i like it more to do a short "return" here
we can remove 1 if, 2 breaks, 1 var with that

Comment (melloware) [reply src/main/java/org/primefaces/renderkit/SelectOneRenderer.java:128]: Hmmm can you elaborate?  I am not quite sure I follow.

Comment (tandraschko) [reply src/main/java/org/primefaces/renderkit/SelectOneRenderer.java:128]: "return findSelectItem(...." but nvm, doesnt work
did a to short review :D

Comment (melloware) [reply src/main/java/org/primefaces/renderkit/SelectOneRenderer.java:128]: Got it. Yeah i only want to break the loop if its found.

**Comment by tandraschko** [comment]
i would still prefer 2 protected methods (findSelectItemByLabel, findSelectItemByValue) and they should call the private one. WDYT?

also it would be great to have 2-3 basic unittests

**Comment by melloware** [comment]
OK renamed methods ByLabel and ByValue and created private method called by both.

I already have Integration Tests for both scenarios CascadeSelect tests the `selectItemByValue` and the SelectOne tests `findItemByLabel` so we will know if this is broken or not.

**Comment by melloware** [comment]
All Integration Tests passing after this fix: https://github.com/primefaces-extensions/primefaces-integration-tests/actions/runs/656557637

**Pull request merged**